octis/optimization/optimizer_evaluation.py

```diff
@@ -21,7 +22,9 @@ def __init__(self, optimizer, BO_results):
         dict_metric_parameters = dict()
 
         for mp in metric_parameters:
-            if check_instance(getattr(optimizer.metric, mp)):
+            if isinstance(getattr(optimizer.metric,mp),(Coherence,TopicDiversity)):
+                pass
+            elif check_instance(getattr(optimizer.metric, mp)):
```

Comment:
what is the difference between `isinstance()` and `check_instance()`?

Reply:
It looks like the changes should be in `check_instance()`, not in `optimizer_evaluation`:

https://github.com/mila-aia/OCTIS/blob/168b83a54c356df4a3fa117e433ba829bb0776ad/octis/optimization/optimizer_tool.py#L268

Reply:
The purpose of `check_instance` is specifically to test if something (likes instances of Coherence or TopicDiversity) should be logged in the json file, or not

Reply:
it looks like there's not even a need to change `check_instance()`, it will automatically not log anything else than str, float, int or bool

Reply:
`check_instance` is a function from OCTIS, defined by: 

def check_instance(obj):
"""
    Check if a specific object con be inserted in the json file.

    :param obj: an object of the optimization to be saved
    :type obj: [str,float, int, bool, etc.]
    :return: 'True' if the object can be inserted in a json file, 'False' otherwise
    :rtype: bool
    """
    types = [str, float, int, bool]

    for t in types:
        if isinstance(obj, t):
            return True

    return False
`isinstance()` in our .py file checks if the attribute is either an instance of Coherence or TopicDiversity

Reply:
> it looks like there's not even a need to change `check_instance()`, it will automatically not log anything else than str, float, int or bool

Yes, you are correct!